src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs

```diff
@@ -721,52 +721,82 @@ void UpdateDimensionProperties()
 			else if (Content is FrameworkElement fe)
 			{
 				var explicitHeight = fe.Height;
+				var extentHeight = 0d;
+				var extentWidth = 0d;
 				if (explicitHeight.IsFinite())
 				{
-					ExtentHeight = explicitHeight;
+					extentHeight = explicitHeight;
 				}
 				else
 				{
 					var canUseActualHeightAsExtent =
 						fe.ActualHeight > 0 &&
 						fe.VerticalAlignment == VerticalAlignment.Stretch;
 
-					ExtentHeight = canUseActualHeightAsExtent ? fe.ActualHeight : fe.DesiredSize.Height;
+					extentHeight = canUseActualHeightAsExtent ? fe.ActualHeight : fe.DesiredSize.Height;
 				}
 
+#if __WASM__
+				// Issue needs to be fixed first for WASM for Bottom Margin missing
+				// Details here: https://github.com/unoplatform/uno/issues/7000
+				ExtentHeight = extentHeight + fe.Margin.Top;
+#else
+				ExtentHeight = extentHeight + fe.Margin.Top + fe.Margin.Bottom;
+#endif
+
+
 				var explicitWidth = fe.Width;
 				if (explicitWidth.IsFinite())
 				{
-					ExtentWidth = explicitWidth;
+					extentWidth = explicitWidth;
 				}
 				else
 				{
 					var canUseActualWidthAsExtent =
 						fe.ActualWidth > 0 &&
 						fe.HorizontalAlignment == HorizontalAlignment.Stretch;
 
-					ExtentWidth = canUseActualWidthAsExtent ? fe.ActualWidth : fe.DesiredSize.Width;
+					extentWidth = canUseActualWidthAsExtent ? fe.ActualWidth : fe.DesiredSize.Width;
 				}
+
+#if __WASM__
+				// Issue needs to be fixed first for WASM for Right Margin missing
+				// Details here: https://github.com/unoplatform/uno/issues/7000
+				ExtentWidth = extentWidth + fe.Margin.Left;
+#else
+				ExtentWidth = extentWidth + fe.Margin.Left + fe.Margin.Right;
```

Comment:
As this is still a problem - should #7000 be closed with this PR?

Reply:
Correctly updated the description of the issue, thanks ;) !